### PR TITLE
docs: security self assessment

### PR DESCRIPTION
**Pull request by jbw976**
### Description of your changes

This PR contains a new security self assessment for the Crossplane project, as required by our open CNCF Graduation [application issue](https://github.com/cncf/toc/issues/1397). This assessment follows the template provided by TAG-Security found at https://tag-security.cncf.io/community/assessments/guide/self-assessment/.

This document is being added to the `security` folder, but could potentially be moved to TAG-Security's repo if they prefer it to live there long term.

@crossplane/crossplane-maintainers please take a look and provide any feedback or clarifications you'd like to see. As the template calls out, it is not a full audit, but we should still be documenting and explaining security relevant aspects of the project to the best of our ability:

> It is not intended to provide a security audit of Crossplane, or function as an independent assessment or attestation of Crossplane’s security health.

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `earthly +reviewable` to ensure this PR is ready for review.
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/main/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/main/contributing#checklist-cheat-sheet

### CODE REVIEW

File: security/self-assessment.md
Comment (phisco) [inline security/self-assessment.md:306]: maybe worth mentioning https://github.com/crossplane/crossplane/security/policy#reporting-a-vulnerability too somewhere here?

Comment (phisco) [reply security/self-assessment.md:306]: I now saw it a few lines down 😅

**Review by phisco — APPROVED** [review]
LGTM, thanks! I'll see if I can find other projects' self-assessments to check if there is anything else worth adding.

**Comment by jbw976** [comment]
Thanks for taking a look @phisco! 🙇‍♂️   

> I'll see if I can find other projects' self-assessments to check if there is anything else worth adding.

One example I found was [cert-manager](https://github.com/cncf/tag-security/blob/1f24478023d92c76372e92e478238a04cafd1658/community/assessments/projects/cert-manager/self-assessment.md)

**Comment by jbw976** [comment]
We got @phisco's blessing on this, so I'm going to go ahead and merge. If others have feedback later on, we can always follow up with another PR 😇

**Pull request merged**
